rpc: Fix addnode remove command error

fjahr opened the pull request:
The `addnode` RPC with the `remove` command parameter is used to remove a node from the "added nodes". It did not have test coverage and in case of failure to remove the node it responded with the confusing message "Error: Node has not been added.".

This PR adds test coverage and introduces a new error code as well as changes the error message to something that makes sense.

fjahr committed:
rpc: Improve addnode remove command error message

This also adds test coverage for the remove command which was uncovered before.

jonatack commented on test/functional/rpc_net.py:
```diff
@@ -133,6 +133,13 @@ def _test_getaddednodeinfo(self):
         assert_equal(added_nodes[0]['addednode'], ip_port)
         # check that a non-existent node returns an error
         assert_raises_rpc_error(-24, "Node has not been added", self.nodes[0].getaddednodeinfo, '1.1.1.1')
+        # check that node can not be added again
```
Perhaps place this test (or all the new ones) two lines higher as it is order-dependent on the addnode call.

s/can not/cannot/

jonatack reviewed (approved):
Tested ACK 9795d93ab445c0e

Feel free to ignore the comments below unless you have to retouch.

MarcoFalke commented:
I think the previous error was fine. Can't remove node, because node has not been added. No strong opinion, though.

Concept ACK on adding test coverage here: https://marcofalke.github.io/btc_cov/total.coverage/src/rpc/net.cpp.gcov.html#274

jonatack commented:
I agree with @fjahr that this error message is confusing. Good idea on adding the missing coverage; thank you for reminding me of your btc_cov link.

MarcoFalke commented:
If the change stays, it needs release notes, because it is a breaking RPC change

promag commented:
Agree with @MarcoFalke, the change in the error isn't worth a breaking change. I'd rather document the error.

On a side note, instead of raising RPC errors, I'd return `{ "message": "node not found" }` or something like that.

fjahr commented:
I was not sure but I assume changing the error code is breaking but changing the error message is not considered breaking? I have changed the error code back and kept an improved error message that should explain the reasoning behind the error code better. Also addressed @jonatack 's nits.

theStack reviewed (approved):
I think the confusing part in the error message was that "not added" could be possibly interpreted here both as an action/verb (i.e. "didn't add node in the course of this RPC call") as well as referring to a state (i.e. "is currently not in the list of added nodes"). It is clearer now that the latter is meant, and increasing test coverage is always a good idea.

Tested ACK https://github.com/bitcoin/bitcoin/commit/a51d0ad2de

luke-jr commented:
IMO changes to the string description shouldn't be considered a breaking change.

luke-jr reviewed (approved):
utACK

laanwj commented:
> IMO changes to the string description shouldn't be considered a breaking change.

I agree here.

MarcoFalke commented:
In a previous version the error code (integer) was also changed. Changing the error code has usually been considered a breaking change
